assets/pages/new-contributions-landing/components/ContributionUsTicker.jsx

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+// ----- Imports ----- //
+import React, { Component } from 'react';
+import { classNameWithModifiers } from 'helpers/utilities';
+
+// ---- Types ----- //
+type StateTypes = {|
+  totalSoFar: number,
+  goal: number,
+|}
+
+type PropTypes = {}
+
+
+// ---- Helpers ----- //
+
+const getInitialTickerValues = (): Promise<{| totalSoFar: number, goal: number |}> =>
+  fetch('https://interactive.guim.co.uk/docsdata-test/1ySn7Ol2NQLvvSw_eAnVrPuuRnaGOxUmaUs6svtu_irU.json')
+    .then(resp => resp.json())
+    .then((data) => {
+      const totalSoFar = parseInt(data.sheets.Sheet1[0].total, 10);
+      const goal = parseInt(data.sheets.Sheet1[0].goal, 10);
+
+      return { totalSoFar, goal };
+    });
+
+const percentageTotalAsNegative = (total: number, goal: number) => {
+  const percentage = ((total / goal) * 100) - 100;
+  return percentage > 0 ? 0 : percentage;
+};
+
+
+// ----- Component ----- //
+export class ContributionUsTicker extends Component<PropTypes, StateTypes> {
+
+  constructor(props: PropTypes) {
+    super(props);
+
+    this.state = {
+      totalSoFar: 0,
+      goal: 0,
+    };
+
+  }
+
+  componentDidMount(): void {
+    const self = this;
+    getInitialTickerValues().then(({ totalSoFar, goal }) => {
+      const initialTotal = totalSoFar * 0.8;
+      this.count = initialTotal;
+      this.totalSoFar = totalSoFar;
+      self.setState({ totalSoFar: initialTotal, goal });
+      window.setTimeout(() => {
+        window.requestAnimationFrame(this.increaseCounter);
+        this.animateBar(totalSoFar, goal);
+      }, 500);
+    });
+  }
+
+  increaseCounter = this.increaseCounter.bind(this);
+  filledProgressBar: ?HTMLDivElement;
+  count = 0;
+  totalSoFar: number;
+
+  animateBar(totalSoFar: number, goal: number) {
+    const progressBarElement = this.filledProgressBar;
+    if (progressBarElement && progressBarElement instanceof HTMLElement) {
+      progressBarElement.style.transform = `translateX(${percentageTotalAsNegative(totalSoFar, goal)}%)`;
+    }
+  }
+
+  increaseCounter() {
```

Comment:
wow, this is a far out way of defining a function! No `const` or `function`. Is this style/syntax specific to stateful components?

Reply:
👍